distributed/shuffle/_shuffle.py

```diff
@@ -49,7 +49,9 @@ def shuffle_transfer(
             input, id, npartitions=npartitions, column=column
         )
     except Exception:
-        raise RuntimeError(f"shuffle_transfer failed during shuffle {id}")
+        msg = f"shuffle_transfer failed during shuffle {id}"
+        logger.error(msg, exc_info=True)
+        raise RuntimeError(msg)
```

Comment:
you should get the proper traceback with

```suggestion
    except Exception as exc:
        raise RuntimeError(msg) from exc
```

Reply:
Exception chaining sometimes threw a recursion error back in #7326 which is why we dropped it.